src/test/java/com/spanish_inquisition/battleship/server/BattleshipServerTestIT.java

```diff
@@ -52,9 +52,39 @@ public void shouldConnectWithTwoPlayers() {
             }
         }).run();
         // When
-        List<Socket> testClients = BattleshipServer.connectWithPlayers(serverSocket);
+        BattleshipServer.connectWithPlayers(serverSocket);
         // Then
-        Assert.assertEquals(2, testClients.size());
+        Assert.assertEquals(2, BattleshipServer.clients.size());
+    }
+
+    @Test
+    public void shouldReadNamesFromClients() throws InterruptedException {
+        // Given
+        new Thread(() -> {
+            try {
+                client1 = new Socket(SERVER_ADDRESS, TEST_PORT);
+                PrintWriter writer = new PrintWriter(client1.getOutputStream(), true);
+                writer.println("Name 1");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }).run();
+        new Thread(() -> {
+            try {
+                client2 = new Socket(SERVER_ADDRESS, TEST_PORT);
+                PrintWriter writer = new PrintWriter(client2.getOutputStream(), true);
+                writer.println("Name 2");
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }).run();
```

Comment:
Method bodies can be extracted as a 

`void assignSocketAndNameTo(Client client, String nameString) {`
`try {`
`    client = new Socket(SERVER_ADDRESS, TEST_PORT);`
`    PrintWriter writer = new PrintWriter(client.getOutputStream(), true);`
`    writer.println(nameString);`
`catch () {}`
`}`
And the thread bodies would be one line